diff --git a/src/sysv/systemd/sssd.service.in b/src/sysv/systemd/sssd.service.in
index 06445ea..b0d7311 100644
--- a/src/sysv/systemd/sssd.service.in
+++ b/src/sysv/systemd/sssd.service.in
@@ -1,7 +1,7 @@
 [Unit]
 Description=System Security Services Daemon
 # SSSD will not be started until syslog is
-After=syslog.target
+# After=syslog.target
 # SSSD must be running before we permit user sessions
 Before=systemd-user-sessions.service nss-user-lookup.target
 Wants=nss-user-lookup.target
